lib/_http_agent.js

```diff
@@ -112,11 +112,15 @@ function Agent(options) {
         if (this.sockets[name])
           count += this.sockets[name].length;
 
-        if (count > this.maxSockets || freeLen >= this.maxFreeSockets) {
+        if (count > this.maxSockets) {
           socket.destroy();
         } else if (this.keepSocketAlive(socket)) {
```

Comment:
I think you can do this with just if & else with `this.keepSocketAlive(socket) && count < this.maxSockets`.